From 5c8c860fc66ffa74967fd28b013cd8725eb9ae6f Mon Sep 17 00:00:00 2001
From: grotaclas <grotaclas@gmail.com>
Date: Sun, 04 May 2025 19:41:51 -0000
Subject: [PATCH] Make Persian mission "Checks and Balance" work with all
 estates

The mission didn't consider dhimmi, brahmins, rajput, vaisyas, jains,
janissaries and eunuchs when dermining if all estates have 40 influence
and two estates have 60 influence.

Bugreport: https://forum.paradoxplaza.com/forum/threads/persian-mission-requirements-met-but-not-completable.1694807/
---
 missions/KoK_Persia_Missions.txt | 132 ++++++++-----------------------
 1 file changed, 32 insertions(+), 100 deletions(-)

diff --git a/missions/KoK_Persia_Missions.txt b/missions/KoK_Persia_Missions.txt
index 2bcbfc3b..2e259caf 100755
--- a/missions/KoK_Persia_Missions.txt
+++ b/missions/KoK_Persia_Missions.txt
@@ -3625,108 +3625,40 @@ kok_persia_culture_mission_slot_4 = {
 			custom_trigger_tooltip = {
 				tooltip = per_checks_and_balance_tt #Two ?Yestates?! have at least ?Y60 influence?!, the other has at least ?Y40 influence?!.
 				calc_true_if = {
-					OR = { NOT = { has_estate = estate_burghers } 		estate_influence = { estate = estate_burghers 		influence = 40 } }
-					OR = { NOT = { has_estate = estate_church } 		estate_influence = { estate = estate_church 		influence = 40 } }
-					OR = { NOT = { has_estate = estate_nobles } 		estate_influence = { estate = estate_nobles 		influence = 40 } }
-					OR = { NOT = { has_estate = estate_qizilbash } 		estate_influence = { estate = estate_qizilbash 		influence = 40 } }
-					OR = { NOT = { has_estate = estate_ghulams } 		estate_influence = { estate = estate_ghulams 		influence = 40 } }
+					OR = { NOT = { has_estate = estate_brahmins }		estate_influence = { estate = estate_brahmins		influence = 40 } }
+					OR = { NOT = { has_estate = estate_rajput }			estate_influence = { estate = estate_rajput			influence = 40 } }
+					OR = { NOT = { has_estate = estate_maratha }		estate_influence = { estate = estate_maratha		influence = 40 } }
+					OR = { NOT = { has_estate = estate_vaisyas }		estate_influence = { estate = estate_vaisyas		influence = 40 } }
+					OR = { NOT = { has_estate = estate_church }			estate_influence = { estate = estate_church			influence = 40 } }
+					OR = { NOT = { has_estate = estate_nobles }			estate_influence = { estate = estate_nobles			influence = 40 } }
+					OR = { NOT = { has_estate = estate_burghers }		estate_influence = { estate = estate_burghers		influence = 40 } }
+					OR = { NOT = { has_estate = estate_cossacks }		estate_influence = { estate = estate_cossacks		influence = 40 } }
 					OR = { NOT = { has_estate = estate_nomadic_tribes }	estate_influence = { estate = estate_nomadic_tribes	influence = 40 } }
-					amount = 6
-				}
-				if = {
-					limit = { has_estate = estate_qizilbash }
-					if = {
-						limit = { has_estate = estate_ghulams }
-						if = {
-							limit = { has_estate = estate_nomadic_tribes }
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								estate_influence = { estate = estate_qizilbash 		influence = 60 }
-								estate_influence = { estate = estate_ghulams 		influence = 60 }
-								estate_influence = { estate = estate_nomadic_tribes	influence = 60 }
-								amount = 2
-							}
-						}
-						else = {
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								estate_influence = { estate = estate_qizilbash 		influence = 60 }
-								estate_influence = { estate = estate_ghulams 		influence = 60 }
-								amount = 2
-							}
-						}
-					}
-					else = {
-						if = {
-							limit = { has_estate = estate_nomadic_tribes }
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								estate_influence = { estate = estate_qizilbash 		influence = 60 }
-								estate_influence = { estate = estate_nomadic_tribes	influence = 60 }
-								amount = 2
-							}
-						}
-						else = {
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								estate_influence = { estate = estate_qizilbash 		influence = 60 }
-								amount = 2
-							}
-						}
-					}
+					OR = { NOT = { has_estate = estate_dhimmi }			estate_influence = { estate = estate_dhimmi			influence = 40 } }
+					OR = { NOT = { has_estate = estate_jains }			estate_influence = { estate = estate_jains			influence = 40 } }
+					OR = { NOT = { has_estate = estate_janissaries }	estate_influence = { estate = estate_janissaries	influence = 40 } }
+					OR = { NOT = { has_estate = estate_eunuchs }		estate_influence = { estate = estate_eunuchs		influence = 40 } }
+					OR = { NOT = { has_estate = estate_qizilbash }		estate_influence = { estate = estate_qizilbash		influence = 40 } }
+					OR = { NOT = { has_estate = estate_ghulams }		estate_influence = { estate = estate_ghulams		influence = 40 } }
+					amount = 15
 				}
-				else = {
-					if = {
-						limit = { has_estate = estate_ghulams }
-						if = {
-							limit = { has_estate = estate_nomadic_tribes }
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								estate_influence = { estate = estate_ghulams 		influence = 60 }
-								estate_influence = { estate = estate_nomadic_tribes	influence = 60 }
-								amount = 2
-							}
-						}
-						else = {
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								estate_influence = { estate = estate_ghulams 		influence = 60 }
-								amount = 2
-							}
-						}
-					}
-					else = {
-						if = {
-							limit = { has_estate = estate_nomadic_tribes }
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								estate_influence = { estate = estate_nomadic_tribes	influence = 60 }
-								amount = 2
-							}
-						}
-						else = {
-							calc_true_if = {
-								estate_influence = { estate = estate_burghers 		influence = 60 }
-								estate_influence = { estate = estate_church 		influence = 60 }
-								estate_influence = { estate = estate_nobles 		influence = 60 }
-								amount = 2
-							}
-						}
-					}
+				calc_true_if = {
+					if = { limit = { has_estate = estate_brahmins }		estate_influence = { estate = estate_brahmins		influence = 60 } }
+					if = { limit = { has_estate = estate_rajput }		estate_influence = { estate = estate_rajput			influence = 60 } }
+					if = { limit = { has_estate = estate_maratha }		estate_influence = { estate = estate_maratha		influence = 60 } }
+					if = { limit = { has_estate = estate_vaisyas }		estate_influence = { estate = estate_vaisyas		influence = 60 } }
+					if = { limit = { has_estate = estate_church }		estate_influence = { estate = estate_church			influence = 60 } }
+					if = { limit = { has_estate = estate_nobles }		estate_influence = { estate = estate_nobles			influence = 60 } }
+					if = { limit = { has_estate = estate_burghers }		estate_influence = { estate = estate_burghers		influence = 60 } }
+					if = { limit = { has_estate = estate_cossacks }		estate_influence = { estate = estate_cossacks		influence = 60 } }
+					if = { limit = { has_estate = estate_nomadic_tribes } estate_influence = { estate = estate_nomadic_tribes influence = 60 } }
+					if = { limit = { has_estate = estate_dhimmi }		estate_influence = { estate = estate_dhimmi			influence = 60 } }
+					if = { limit = { has_estate = estate_jains }		estate_influence = { estate = estate_jains			influence = 60 } }
+					if = { limit = { has_estate = estate_janissaries }	estate_influence = { estate = estate_janissaries	influence = 60 } }
+					if = { limit = { has_estate = estate_eunuchs }		estate_influence = { estate = estate_eunuchs		influence = 60 } }
+					if = { limit = { has_estate = estate_qizilbash }	estate_influence = { estate = estate_qizilbash		influence = 60 } }
+					if = { limit = { has_estate = estate_ghulams }		estate_influence = { estate = estate_ghulams		influence = 60 } }
+					amount = 2
 				}
 			}
 			OR = {
-- 
2.45.2

